rotkehlchen/exchanges/ftx.py

```diff
@@ -0,0 +1,514 @@
+                f'{response.status_code} and text: {response.text}',
+            )
+
+        try:
+            json_ret = jsonloads_dict(response.text)
+        except JSONDecodeError as e:
+            raise RemoteError(f'FTX returned invalid JSON response: {response.text}') from e
+
+        if 'result' not in json_ret:
+            raise RemoteError(f'FTX json response does not contain data: {response.text}')
+
+        return json_ret['result']
+
+    @overload
+    def _api_query(  # pylint: disable=no-self-use
+            self,
+            endpoint: Literal['wallet/all_balances'],
+            start_time: Optional[Timestamp] = None,
+            end_time: Optional[Timestamp] = None,
+            limit: int = PAGINATION_LIMIT,
+            paginate: bool = True,
+    ) -> Dict[str, List[Any]]:
+        ...
+
+    @overload
+    def _api_query(
+            self,
+            endpoint: Literal[
+                'fills',
+                'wallet/deposits',
+                'wallet/withdrawals',
+                'markets',
+                'markets/eth/eur/trades',
+            ],
+            start_time: Optional[Timestamp] = None,
+            end_time: Optional[Timestamp] = None,
+            limit: int = PAGINATION_LIMIT,
+            paginate: bool = True,
+    ) -> List[Dict[str, Any]]:
+        ...
+
+    def _api_query(
+            self,
+            endpoint: str,
+            start_time: Optional[Timestamp] = None,
+            end_time: Optional[Timestamp] = None,
+            limit: int = PAGINATION_LIMIT,
+            paginate: bool = True,
+    ) -> Union[List[Dict[str, Any]], Dict[str, List[Any]]]:
+        """Query FTX endpoint and retrieve all available information if pagination
+        is requested. In case of paginate being set to False only one request is made.
+        Can raise:
+        - RemoteError
+        """
+        if not paginate:
+            final_data_no_pag = self._make_request(
+                endpoint=endpoint,
+                limit=limit,
+                start_time=start_time,
+                end_time=end_time,
+            )
+            return final_data_no_pag
+
+        # If there is pagination we follow the example from the official ftx python example
+        # https://github.com/ftexchange/ftx/blob/master/rest/client.py#L163
+        # In this case the strategy is a while loop leaving fixed the start_time (lower bound)
+        # and decreasing end time (the upper bound) until we fetch all the available information
+        new_end_time = end_time
+        ids = set()
+        final_data: List[Dict[str, Any]] = []
+        while True:
+            step = self._make_request(
+                endpoint=endpoint,
+                limit=limit,
+                start_time=start_time,
+                end_time=new_end_time,
+            )
+
+            assert isinstance(step, list), 'Doing pagination in something that is not a list'
```

Comment:
Remove the assertion from here. Change to a remote error. Assertions are for things that should never ever happen. If an assertion hits, it should only be due to a programming error.

This here can happen from external data. If for some reason the endpoint does not return a list. So change to an if statement that raises a RemoteError